Update Invoices setting screen

rezkiy37 opened the pull request:
<!-- If necessary, assign reviewers that know the area or changes well. Feel free to tag any additional reviewers you see fit. -->

### Details
<!-- Explanation of the change or anything fishy that is going on -->
The PR replaces the legacy sections of the invoices page with the new ones. 
**Note:** this PR should be merged right after https://github.com/Expensify/App/pull/50571 because it integrates API there.

### Fixed Issues
<!---
1. Please postfix `$` with a URL link to the GitHub issue this Pull Request is fixing. For example, `$ https://github.com/Expensify/App/issues/<issueID>`.
2. Please postfix  `PROPOSAL:` with a URL link to your GitHub comment, which contains the approved proposal (i.e. the proposal that was approved by Expensify).  For example, `PROPOSAL: https://github.com/Expensify/App/issues/<issueID>#issuecomment-1369752925`

Do NOT add the special GH keywords like `fixed` etc, we have our own process of managing the flow.
It MUST be an entire link to the github issue and your comment proposal ; otherwise, the linking and its automation will not work as expected.

Make sure this section looks similar to this (you can link multiple issues using the same formatting, just add a new line):

$ https://github.com/Expensify/App/issues/<issueID>
$ https://github.com/Expensify/App/issues/<issueID(comment)>

Do NOT only link the issue number like this: $ #<issueID>
--->
$ https://github.com/Expensify/App/issues/45175
PROPOSAL: N/A


### Tests
<!---
Add a numbered list of manual tests you performed that validates your changes work on all platforms, and that there are no regressions present.
Add any additional test steps if test steps are unique to a particular platform.
Manual test steps should be written so that your reviewer can repeat and verify one or more expected outcomes in the development environment.

For example:
1. Click on the text input to bring it into focus
2. Upload an image via copy paste
3. Verify a modal appears displaying a preview of that image
--->

Preconditions:
- Enable invoices of a workspace

1. Open the workspace’s invoices page
2. Verify that there are 2 sections: balance, bank accounts and details.

- [x] Verify that no errors appear in the JS console

### Offline tests
<!---
Add any relevant steps that validate your changes work as expected in a variety of network states e.g. "offline", "spotty connection", "slow internet", etc. Manual test steps should be written so that your reviewer and QA testers can repeat and verify one or more expected outcomes. If you are unsure how the behavior should work ask for advice in the `#expensify-open-source` Slack channel.
--->

Same as "Tests".

### QA Steps
<!---
Add a numbered list of manual tests that can be performed by our QA engineers on the staging environment to validate that your changes work on all platforms, and that there are no regressions present.
Add any additional QA steps if test steps are unique to a particular platform.
Manual test steps should be written so that the QA engineer can repeat and verify one or more expected outcomes in the staging environment.

For example:
1. Click on the text input to bring it into focus
2. Upload an image via copy paste
3. Verify a modal appears displaying a preview of that image
--->

Same as "Tests".

- [x] Verify that no errors appear in the JS console

### PR Author Checklist
<!--
This is a checklist for PR authors. Please make sure to complete all tasks and check them off once you do, or else your PR will not be merged!
-->

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

<!-- add screenshots or videos here -->
<img width="472" alt="Android" src="https://github.com/user-attachments/assets/0da23229-4a21-4721-8900-1f7b2a0ddd0f">

</details>

<details>
<summary>Android: mWeb Chrome</summary>

<!-- add screenshots or videos here -->
<img width="470" alt="Android Chrome" src="https://github.com/user-attachments/assets/deba5301-913f-40df-a8e5-47e6afb8acba">

</details>

<details>
<summary>iOS: Native</summary>

<!-- add screenshots or videos here -->
<img width="470" alt="IOS" src="https://github.com/user-attachments/assets/58307b83-0474-40fb-9566-f8765a4f78d6">

</details>

<details>
<summary>iOS: mWeb Safari</summary>

<!-- add screenshots or videos here -->
<img width="441" alt="IOS Simulator" src="https://github.com/user-attachments/assets/66f5e526-b547-4220-bb59-91bfc2c0c976">

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

<!-- add screenshots or videos here -->
<img width="1728" alt="Chrome" src="https://github.com/user-attachments/assets/845d13f7-7cc9-4316-a134-f84fdb5897dc">

</details>

<details>
<summary>MacOS: Desktop</summary>

<!-- add screenshots or videos here --><img width="1217" alt="Desktop" src="https://github.com/user-attachments/assets/4df16bb0-a9b0-4af6-8bdf-e5c88d7e980b">

</details>

rezkiy37 committed:
remove old components

rezkiy37 committed:
add WorkspaceInvoicingDetailsSection

rezkiy37 committed:
Merge branch 'main' of https://github.com/rezkiy37/Expensify into feature/45175-invoicing-settings-screen

rezkiy37 committed:
Merge branch 'main' of https://github.com/rezkiy37/Expensify into feature/45175-invoicing-settings-screen

rezkiy37 committed:
Merge branch 'main' of https://github.com/rezkiy37/Expensify into feature/45175-invoicing-settings-screen

shawnborton commented:
We're still missing the stylized page header:
![CleanShot 2024-10-17 at 17 19 11@2x](https://github.com/user-attachments/assets/6d663d3f-fbf1-4e63-9b39-0b526fc5746d)

rezkiy37 commented:
> **Note:** this PR should be merged right after https://github.com/Expensify/App/pull/50571 because it integrates API there.

rezkiy37 committed:
apply icon

rezkiy37 commented:
> We're still missing the stylized page header: ![CleanShot 2024-10-17 at 17 19 11@2x](https://private-user-images.githubusercontent.com/2319350/377529522-6d663d3f-fbf1-4e63-9b39-0b526fc5746d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjkxODMxNTMsIm5iZiI6MTcyOTE4Mjg1MywicGF0aCI6Ii8yMzE5MzUwLzM3NzUyOTUyMi02ZDY2M2QzZi1mYmYxLTRlNjMtOWIzOS0wYjUyNmZjNTc0NmQucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTAxNyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEwMTdUMTYzNDEzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZTg0ZDczY2YzMjgwZmM2NTA1NDA5ZTRjMTY2YTU3NzBlODE1NGUwZDNkMTFhNDE0NzFmMmJhZTc1MjY5NWRmNCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.FDPrRxQs9wg1Khg-wvte3_KbnrhGQmgtUm16i_UQuFc)

I've added the icon and aligned the header. Fixed - 52d02d8. 

<details><summary>Details</summary>

<img width="1728" alt="Screenshot 2024-10-17 at 18 39 34" src="https://github.com/user-attachments/assets/9e480a10-2cf4-4e6f-baac-cd99ece84b15">

</details> 

cc @shawnborton

shawnborton commented:
Looks good, though I think that's the icon for Bills and not Invoices. Can you swap them out quickly?

shawnborton commented:
I deleted my last comment, you have the right icon in place - sorry about that!

shawnborton commented:
That looks good to me, cc @JmillsExpensify to make sure you have visibility into this one as well.

rezkiy37 commented:
> I deleted my last comment, you have the right icon in place - sorry about that!

Hm... I've found this icon in Figma.

<img width="674" alt="Screenshot 2024-10-17 at 18 51 42" src="https://github.com/user-attachments/assets/fa3769ce-23ae-4e44-84d1-fac5ad3b61d2">

rezkiy37 committed:
Merge branch 'main' of https://github.com/rezkiy37/Expensify into feature/45175-invoicing-settings-screen

rezkiy37 committed:
tweak sections descriptions

rezkiy37 committed:
align test styles

rezkiy37 commented:
Aligned text styles - 71b7feb.

<details><summary>Details</summary>
<p>

<img width="1724" alt="Screenshot 2024-10-22 at 13 13 21" src="https://github.com/user-attachments/assets/c0b1d8d8-a5f7-4792-a978-fb6e366f52a1">


</p>
</details> 

cc @shawnborton

shawnborton commented:
Hmm I'm not seeing any changes? This part of the invoice view:
![CleanShot 2024-10-22 at 13 22 16@2x](https://github.com/user-attachments/assets/ffcbeb68-fd46-4a1d-a81b-92ee58e598c5)

Should look more like the Settings > Wallet view:
![CleanShot 2024-10-22 at 13 22 40@2x](https://github.com/user-attachments/assets/90122893-aa28-4d43-83d8-95dc9932128c)

shawnborton commented:
Though we chatted on Slack that we want all of the paragraph text below the card's title to use our text supporting color, just like the rest of the workspace editor has:

![CleanShot 2024-10-22 at 13 23 24@2x](https://github.com/user-attachments/assets/1054598b-138d-4fd8-a9f8-f3316cceb8c7)

So that means we need to update Settings to match (or not, we can do that as a follow up).

rezkiy37 committed:
use muted descriptions

rezkiy37 commented:
> Hmm I'm not seeing any changes? This part of the invoice view: ![CleanShot 2024-10-22 at 13 22 16@2x](https://private-user-images.githubusercontent.com/2319350/378816804-ffcbeb68-fd46-4a1d-a81b-92ee58e598c5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjk1OTY2MzksIm5iZiI6MTcyOTU5NjMzOSwicGF0aCI6Ii8yMzE5MzUwLzM3ODgxNjgwNC1mZmNiZWI2OC1mZDQ2LTRhMWQtYTgxYi05MmVlNThlNTk4YzUucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTAyMiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEwMjJUMTEyNTM5WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YjQ3NDEyZDE0MjQ2MzllMDMwMDUwOTc4ZDZmNTg3YWE0Yjc1MzI2MmZlMThjNWU5NmRhZmQ4ODc4YTljMmM5ZiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.MASyIBxHy0qL5C8FXCnG565WeM1hZqIl15vCGDTW6QI)
> 
> Should look more like the Settings > Wallet view: ![CleanShot 2024-10-22 at 13 22 40@2x](https://private-user-images.githubusercontent.com/2319350/378816895-90122893-aa28-4d43-83d8-95dc9932128c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjk1OTY2MzksIm5iZiI6MTcyOTU5NjMzOSwicGF0aCI6Ii8yMzE5MzUwLzM3ODgxNjg5NS05MDEyMjg5My1hYTI4LTRkNDMtODNkOC05NWRjOTkzMjEyOGMucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTAyMiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEwMjJUMTEyNTM5WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YmM0MzMwY2EwMTRmMDU1ZjI1NjU5MGQxZmQyZjJjNzVmODQ3ZGIwNDRmZDRlMWY0Zjk2NTVjNmQ0NjM4YmIzNyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.cSRFjNdt2tQuVEIHK-7I8LxXq1KGTLNFOQSa1EJtKCo)

Do you think it looks correct?

<img width="1726" alt="Screenshot 2024-10-22 at 13 31 06" src="https://github.com/user-attachments/assets/a5a8aceb-a665-4b80-a0ff-e131df8de3e0">

rezkiy37 committed:
change balance view

rezkiy37 committed:
Merge branch 'main' of https://github.com/rezkiy37/Expensify into feature/45175-invoicing-settings-screen

shawnborton commented:
Nice, that looks good to me!

rezkiy37 commented:
@fedirjh bump

fedirjh reviewed (approved):
LGTM. cc @Expensify/design

shawnborton commented:
Looks pretty good to me! I'm going to run a test build so I can take it for a spin in my account but the screenshots seem correct at first glance.

dannymcclain commented:
Looking much better

shawnborton commented:
Styles look good to me. Is it known that the Company name and Company website fields don't work/don't save though? Not sure if that was a regression introduced here or not.

shawnborton commented:
Sounds good, thanks for confirming. I think this is all good from a design standpoint then, thanks for the changes here!

OSBotify commented:
:hand: This PR was not deployed to staging yet because QA is ongoing. It will be automatically deployed to staging after the next production release.

madmax330 commented:
Tests were passing
